### ISSUE TITLE
java.lang.IllegalArgumentException - removeObserver can only be called after addObserver is called

### ISSUE THREAD

**Issue by ziga-hvalec** [googlemaps/react-native-navigation-sdk#310]
we're getting this error on 0.6.1 on android only

```javascript
const navigationCallbacks: NavigationCallbacks = useMemo(
    () => ({
      onArrival,
      onNavigationReady,
      onNavigationInitError,
      onLocationChanged,
      onRouteStatusResult,
      onStartGuidance,
      onRemainingTimeOrDistanceChanged,
    }),
    [
      onArrival,
      onNavigationReady,
      onNavigationInitError,
      onLocationChanged,
      onRouteStatusResult,
      onStartGuidance,
      onRemainingTimeOrDistanceChanged,
    ],
  );

  useEffect(() => {
    removeListeners(navigationCallbacks);
    addListeners(navigationCallbacks);
    return () => {
      removeListeners(navigationCallbacks);
    };
  }, [navigationCallbacks, addListeners, removeListeners]);
```

**Comment by caio1985** [comment]
Thanks @ziga-hvalec for reporting this. Can you please add more details to the error that you're getting? I see that you have pasted a code snippet but I couldn't find the error associated.

@illuminati1911 once @ziga-hvalec provides the error details, can you please take a look into this? @jokerttu FYI.

**Comment by illuminati1911** [comment]
@ziga-hvalec I can see here that you are removing listeners before adding them:
```tsx
useEffect(() => {
  removeListeners(navigationCallbacks);
  addListeners(navigationCallbacks);
  return () => {
    removeListeners(navigationCallbacks);
  };
}, [navigationCallbacks, addListeners, removeListeners]);
```

Maybe try removing the first line?
```tsx
useEffect(() => {
  addListeners(navigationCallbacks);
  return () => {
    removeListeners(navigationCallbacks);
  };
}, [navigationCallbacks, addListeners, removeListeners]);
```

**Comment by ziga-hvalec** [comment]
But I copied this from your example project. I checked again and this is no longer there? is this now the correct way of adding/removing listeners?

```javascript
useEffect(() => {
  addListeners(navigationCallbacks);
  return () => {
    removeListeners(navigationCallbacks);
  };
}, [navigationCallbacks, addListeners, removeListeners]);
```

Can this be better documented please?

**Comment by ziga-hvalec** [comment]
Will just addingListeners also update them? 
In case navigationCallbacks change?

**Comment by illuminati1911** [comment]
> But I copied this from your example project. I checked again and this is no longer there? is this now the correct way of adding/removing listeners?
> 
Do you know which release had it? It's certainly possible there may have been a bug in the example app that has been fixed.  The correct way should be as it's now in the example app so like this:
```tsx
useEffect(() => {
  addListeners(navigationCallbacks);
  return () => {
    removeListeners(navigationCallbacks);
  };
}, [navigationCallbacks, addListeners, removeListeners]);
```

Let me know if it fixes the issue for you!

> Can this be better documented please?
Yes. The SDK is currently in its beta phase, but as it matures, the documentation should provide more comprehensive guidance."

**Comment by jokerttu** [comment]
We will improve documenation how to handle navigation on next update for README.md
Listener handling will be improved on PR https://github.com/googlemaps/react-native-navigation-sdk/pull/294 after it is finalized and tested internally.

**Comment by jokerttu** [comment]
Closed as #294 is landed fixing the root issue.
Documentation will be updated on separate PR.
